src/main/java/ch/njol/skript/expressions/ExprMessage.java

```diff
@@ -144,37 +222,44 @@ void set(Event e, String message) {
 			}
 			
 		},
-		UNKNOWNCOMMAND("unknown command", "unknown (command|cmd) message", Skript.classExists("org.bukkit.event.command.UnknownCommandEvent"), "org.bukkit.event.command.UnknownCommandEvent"){
+		UNKNOWNCOMMAND("unknown command", "unknown (command|cmd) message", Skript.classExists("org.bukkit.event.command.UnknownCommandEvent")) {
+			@Override
+			Class<? extends Event>[] getEvents() {
+				return new Class[]{UnknownCommandEvent.class};
+			}
+			
 			@Nullable
 			@Override
 			String get(Event e) {
-				if(e instanceof UnknownCommandEvent) return ((UnknownCommandEvent) e).getMessage();
+				if (e instanceof UnknownCommandEvent) return ((UnknownCommandEvent) e).getMessage();
 				return null;
 			}
 			
 			@Override
 			void set(Event e, String message) {
-				if(e instanceof  UnknownCommandEvent) ((UnknownCommandEvent) e).setMessage(message);
+				if (e instanceof UnknownCommandEvent) ((UnknownCommandEvent) e).setMessage(message);
 			}
 		};
 		
+		static String[] patterns;
+		static MessageType[] supportedTypes;
+		
+		static {
+			supportedTypes = Arrays.stream(values()).filter(messageType -> messageType.supported).toArray(MessageType[]::new);
```

Comment:
instead of using a stream why not use an EnumSet and a List
```
supportedTypes = new EnumSet();
patterns = new ArrayList<String>();
MessageType[] types = values();
for (int i = 0; i < types.length; i++) {
    if (types[i].supported) {
        supportedTypes.add(types[i]);
        patterns.add(types[i].pattern);
    }
}
```

Reply:
What are the benefits of EnumSet's and Lists over streams?

Reply:
they are more memory efficient, and it seems like the only advantage of streams in this scenario is that it allows you to cram everything into one line which is arguably not an advantage

Reply:
streams are great but look at [how much overhead is involved in making them work](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/stream/ReferencePipeline.java). in this case, i don't think you are getting enough out of them to justify that overhead

Reply:
Okay, thanks